kong/runloop/balancer.lua

```diff
@@ -291,11 +291,21 @@ do
       if not healthcheck then
         healthcheck = require("resty.healthcheck") -- delayed initialization
       end
+
+      -- Do not run active healthchecks in `stream` module
+      local checks = upstream.healthchecks
+      if ngx.config.subsystem == "stream" then
+        checks = pl_tablex.deepcopy(checks)
+        checks.active.healthy.interval = 0
+        checks.active.unhealthy.interval = 0
+      end
```

Comment:
Note: If we ever allow the stream module to be enabled without enabling the HTTP module (we currently always enable the HTTP module), then we would need to allow the inverse of this logic.